src/runtime/uv/udp.cpp

```diff
@@ -0,0 +1,461 @@
+/*
+Copyright (c) 2025 Lean FRO, LLC. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Sofia Rodrigues
+*/
+
+#include "runtime/uv/udp.h"
+#include <cstring>
+
+namespace lean {
+
+#ifndef LEAN_EMSCRIPTEN
+
+// =======================================
+// Utility functions
+
+// Stores all the things needed to send data to a UDP socket.
+typedef struct {
+    lean_object *promise;
+    lean_object *data;
+    lean_object *socket;
+} udp_send_data;
+
+void lean_uv_udp_socket_finalizer(void* ptr) {
+    lean_uv_udp_socket_object* udp_socket = (lean_uv_udp_socket_object*)ptr;
+
+    lean_always_assert(udp_socket->m_promise_read == nullptr);
+    lean_always_assert(udp_socket->m_byte_array == nullptr);
+
+    udp_socket->m_uv_udp->data = ptr;
+
+    event_loop_lock(&global_ev);
+
+    uv_close((uv_handle_t*)udp_socket->m_uv_udp, [](uv_handle_t* handle) {
+        lean_uv_udp_socket_object* udp_socket = (lean_uv_udp_socket_object*)handle->data;
+        free(udp_socket->m_uv_udp);
+        free(udp_socket);
+    });
+
+    event_loop_unlock(&global_ev);
+}
+
+void initialize_libuv_udp_socket() {
+    g_uv_udp_socket_external_class = lean_register_external_class(lean_uv_udp_socket_finalizer, [](void* obj, lean_object* f) {
+        lean_uv_udp_socket_object* udp_socket = (lean_uv_udp_socket_object*)obj;
+
+        if (udp_socket->m_promise_read != nullptr) {
+            lean_inc(f);
+            lean_apply_1(f, udp_socket->m_promise_read);
+        }
+
+        if (udp_socket->m_byte_array != nullptr) {
+            lean_inc(f);
+            lean_apply_1(f, udp_socket->m_byte_array);
+        }
+    });
+}
+
+// =======================================
+// UDP Socket Operations
+
```

Comment:
All the C++ functions in this file are missing their respective type annotations and also make mistakes with `b_obj_arg` like in tcp.cpp.